<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>cnv</title>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">cnv</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(oncoPredict)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#This script provides an example of how to download cnv (copy number variation) data from the GDC database for GBM #(glioblastoma), how to apply the map_cnv() function to that data to map cnvs to genes, and how to test the drugs in your drug #response dataset to each cnv to identify biomarkers that enrich for drug response.</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#First, download CNV data for your cancer of interest from GDC database. The cnv data will be exported to your #working directory as cnv.txt </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#This code will export the CNV data into a text file called, 'cnv.txt', containing a table with colnames() 'Sample', 'Chromosome', 'Start', 'End', 'Num Probes', 'Segment_Mean'</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">#The genome of reference is hg19.</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">#query.gbm.nocnv&lt;-GDCquery(project = &quot;TCGA-GBM&quot;,</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">#                          data.category = &quot;Copy number variation&quot;,</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">#                          legacy = TRUE,</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="co">#                          file.type = &quot;nocnv_hg19.seg&quot;,</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="co">#                          sample.type = c(&quot;Primary Tumor&quot;))</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">#patient_total&lt;-nrow((query.gbm.nocnv$results)[[1]]) #The total number of patients GDC has CNV data for</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="co">#query.gbm.nocnv$results[[1]]&lt;-query.gbm.nocnv$results[[1]][1:patient_total,]</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co">#GDCdownload(query.gbm.nocnv, files.per.chunk = 100)</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co">#gbm.nocnv&lt;-GDCprepare(query.gbm.nocnv, save = TRUE, save.filename = &quot;GBMnocnvhg19.rda&quot;)</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#write.table(gbm.nocnv, file='cnv.txt')</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="co">#Second, apply map_cnv() to map cnv data to genes. The mapped cnv data will be exported to your working directory as map.RData</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#The mapping is accomplished by intersecting the gene with the overlapping CNV level. If the gene isn't fully #captured by the CNV, an NA will be assigned.</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"></a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="co">#Determine the parameters of the map_cnv() function.</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24">Cnvs&lt;-<span class="kw">read.table</span>(<span class="st">'cnv.txt'</span>, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb1-25" data-line-number="25"></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">#Third, apply idwas() to test each cnv and each drug. The p-values and beta-values for each test will be exported to #your working directory as CnvTestOutput_pVals.txt and CnvTestOutput_betas.txt</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">#Determine the parameters of the idwas() function...</span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">#Set the drug_prediction parameter.</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">#Make sure rownames() are samples, and colnames() are drugs. Also make sure this data is a data frame.</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32">drug_prediction&lt;-<span class="kw">t</span>(<span class="kw">as.data.frame</span>(<span class="kw">read.table</span>(<span class="st">'DrugPredictions.txt'</span>, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="dv">1</span>)))</a>
<a class="sourceLine" id="cb1-33" data-line-number="33"><span class="co">#dim(drug_prediction) #165 198</span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="co">#In this example, I had to replace the '.' in the names of these TCGA samples with '-' so that they are of the same form as samples in the cnv  data (you may not have to do this).</span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="kw">rownames</span>(drug_prediction)&lt;-<span class="kw">gsub</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">rownames</span>(drug_prediction), <span class="dt">fixed=</span>T)</a>
<a class="sourceLine" id="cb1-37" data-line-number="37"></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="co">#Make sure the sample identifiers in the 'drug prediction' data are of similar form as the sample identifiers in the 'data' parameter.</span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39">rows=<span class="kw">rownames</span>(drug_prediction)</a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="kw">rownames</span>(drug_prediction)&lt;-<span class="kw">substring</span>(rows, <span class="dv">3</span>, <span class="kw">nchar</span>(rows))</a>
<a class="sourceLine" id="cb1-41" data-line-number="41">drug_prediction&lt;-<span class="kw">as.data.frame</span>(drug_prediction)</a>
<a class="sourceLine" id="cb1-42" data-line-number="42"></a>
<a class="sourceLine" id="cb1-43" data-line-number="43"><span class="co">#Determine the number of samples you want the CNVs to be amplified in. The default is 10.</span></a>
<a class="sourceLine" id="cb1-44" data-line-number="44">n=<span class="dv">10</span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"></a>
<a class="sourceLine" id="cb1-46" data-line-number="46"><span class="co">#Indicate whether or not you would like to test cnv data. If TRUE, you will test cnv data. If FALSE, you will test mutation data.</span></a>
<a class="sourceLine" id="cb1-47" data-line-number="47">cnv=<span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb1-48" data-line-number="48"></a>
<a class="sourceLine" id="cb1-49" data-line-number="49">original&lt;-<span class="kw">getwd</span>()</a>
<a class="sourceLine" id="cb1-50" data-line-number="50">wd&lt;-<span class="kw">tempdir</span>()</a>
<a class="sourceLine" id="cb1-51" data-line-number="51">savedir&lt;-<span class="kw">setwd</span>(wd)</a>
<a class="sourceLine" id="cb1-52" data-line-number="52"></a>
<a class="sourceLine" id="cb1-53" data-line-number="53"><span class="co">#Apply map_cnv()</span></a>
<a class="sourceLine" id="cb1-54" data-line-number="54"><span class="co">#This function produces the file map.RData, which stores the object 'theCnvQuantVecList_mat'</span></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"><span class="co">#map_cnv(Cnvs=Cnvs)</span></a>
<a class="sourceLine" id="cb1-56" data-line-number="56"></a>
<a class="sourceLine" id="cb1-57" data-line-number="57"><span class="co">#Set the data parameter.</span></a>
<a class="sourceLine" id="cb1-58" data-line-number="58"><span class="co">#load('map.RData') #This loads the object 'theCnvQuantVecList_mat', which was obtained using map_cnv()</span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="co">#Make sure this data is a data frame and that colnames() are samples.</span></a>
<a class="sourceLine" id="cb1-60" data-line-number="60"><span class="co">#data&lt;-as.data.frame(t(theCnvQuantVecList_mat))</span></a>
<a class="sourceLine" id="cb1-61" data-line-number="61"><span class="co">#samps&lt;-colnames(data)</span></a>
<a class="sourceLine" id="cb1-62" data-line-number="62"><span class="co">#colnames(data)&lt;-substr(samps,1,nchar(samps)-12)</span></a>
<a class="sourceLine" id="cb1-63" data-line-number="63"></a>
<a class="sourceLine" id="cb1-64" data-line-number="64"><span class="co">#library(TxDb.Hsapiens.UCSC.hg19.knownGene)</span></a>
<a class="sourceLine" id="cb1-65" data-line-number="65"><span class="co">#library(org.Hs.eg.db)</span></a>
<a class="sourceLine" id="cb1-66" data-line-number="66"><span class="co">#library(maftools)</span></a>
<a class="sourceLine" id="cb1-67" data-line-number="67"></a>
<a class="sourceLine" id="cb1-68" data-line-number="68"><span class="co">#Apply idwas()</span></a>
<a class="sourceLine" id="cb1-69" data-line-number="69"><span class="co">#idwas(drug_prediction=drug_prediction, data=data, n=n, cnv=cnv)</span></a>
<a class="sourceLine" id="cb1-70" data-line-number="70"></a>
<a class="sourceLine" id="cb1-71" data-line-number="71"><span class="kw">setwd</span>(original)</a></code></pre></div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
